### PR TITLE
Features/slider state

### PR DESCRIPTION
**Pull request by gcamacho079**
Add wrapAround property to config. Build out getSlideIndex function to return correct index based on change amount and config.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,8 +19,41 @@ const bestSlider = (args = {}) => {
...
+    let newIndex;
+
+    if (newIndexIsOutsideOfRange(tempIndex)) {
+      if (wrapAround) {
+        if (tempIndex < 1) {
+          newIndex = totalSlides;
+        } else {
+          newIndex = 1;
+        }
+      } else {
```
Comment (MrSpecific) [inline src/index.js]: @gcamacho079 - I was thinking we should allow changeAmounts of more than just +-1, like the ability to jump 3 slides at a time, for example - so I was envisioning some logic to take that into account when crossing the boundaries of the slider. Thoughts?

**Review by MrSpecific — APPROVED** [review]
TRUE BLUE HERO!

File: src/index.js (hunk 1 above)
Comment (gcamacho079) [reply src/index.js]: Oh absolutely makes sense. I'll revisit this

**Comment by gcamacho079** [comment]
@MrSpecific I've added a new function for changing slides by a variable amount + adjusted the existing function to return the correct slide index

File: src/index.js [hunk 2]
```diff
@@ -23,24 +23,23 @@ const bestSlider = (args = {}) => {
...
   };
 
   // Get new slide index based on change amount
   const getSlideIndex = (changeAmount) => {
     const { activeSlide, totalSlides } = state;
     const { wrapAround } = config;
 
-    let tempIndex = activeSlide + changeAmount;
+    // Take into account change values that would result in multiple wraparounds
+    const adjustedChangeAmount = changeAmount % totalSlides;
```
Comment (MrSpecific) [inline src/index.js:35]: 😍

Comment (gcamacho079) [reply src/index.js:35]: 🥳 🎉

**Pull request merged**